x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/googlevertexai/embeddings/GoogleVertexAiEmbeddingsTaskSettingsTests.java

```diff
@@ -141,16 +259,29 @@ protected GoogleVertexAiEmbeddingsTaskSettings mutateInstanceForVersion(
     }
 
     private static GoogleVertexAiEmbeddingsTaskSettings createRandom() {
-        return new GoogleVertexAiEmbeddingsTaskSettings(randomFrom(new Boolean[] { null, randomBoolean() }));
+        var inputType = randomBoolean() ? randomWithoutUnspecified() : null;
```

Comment:
This class as a `mutateInstanceForVersion` function that covers the serialisation changes you made. Modify this method to check that when writing to an older node (pre TransportVersions.VERTEX_AI_INPUT_TYPE_ADDED) then in inputType field is not written and is null. Something like

```
    protected GoogleVertexAiEmbeddingsTaskSettings mutateInstanceForVersion(
        GoogleVertexAiEmbeddingsTaskSettings instance,
        TransportVersion version
    ) {
        if (version.before(TransportVersions.VERTEX_AI_INPUT_TYPE_ADDED)) {
            return new GoogleVertexAiEmbeddingsTaskSettings(instance.autoTruncate(), null)
        }
        return instance;
    }
```